twisted/test/test_compat.py

```diff
@@ -885,3 +885,25 @@ def write(self, data):
         self.patch(sys, "stdout", stdout)
         self.assertEqual(raw_input("Prompt"), "User input")
         self.assertEqual(stdout.data, "Prompt")
+
+
+class FutureBytesReprTests(unittest.TestCase):
+    """
+    Tests for L{twisted.python.compat._bytesRepr}.
+    """
+
+    def test_bytesReprNotBytes(self):
+        """
+        L{twisted.python.compat._bytesRepr} raises a
+        L{TypeError} when called any object that is not an instance of
+        L{bytes}.
+        """
+        exc = self.assertRaises(TypeError, _bytesRepr, ["not bytes"])
+        self.assertEquals(str(exc), "Expected bytes not ['not bytes']")
+
```

Comment:
You should run twistedchecker on this file.  It will complain that you need two newlines here, I think.